iam/api-client/custom_roles.py

```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/env python
+
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Demonstrates how to perform basic operations with Google Cloud IAM
+custom roles.
+
+For more information, see the documentation at
+https://cloud.google.com/iam/docs/creating-custom-roles.
+"""
+
+import argparse
+import os
+
+from google.oauth2 import service_account
+import googleapiclient.discovery
+
+credentials = service_account.Credentials.from_service_account_file(
+    filename=os.environ['GOOGLE_APPLICATION_CREDENTIALS'],
+    scopes=['https://www.googleapis.com/auth/cloud-platform'])
+service = googleapiclient.discovery.build(
+    'iam', 'v1', credentials=credentials)
+
+
+# [START iam_query_testable_permissions]
+def query_testable_permissions(resource):
+    """Lists valid permissions for a resource."""
+
+    # pylint: disable=no-member
+    permissions = service.permissions().queryTestablePermissions(body={
+        'fullResourceName': resource
+    }).execute()['permissions']
+    for p in permissions:
+        print(p['name'])
+# [END iam_query_testable_permissions]
+
+
+# [START iam_get_role]
+def get_role(name):
+    """Gets a role."""
+
+    # pylint: disable=no-member
+    role = service.roles().get(name=name).execute()
+    print(role['name'])
+    for permission in role['includedPermissions']:
+        print(permission)
+# [END iam_get_role]
+
+
+# [START iam_create_role]
+def create_role(name, project, title, description, permissions, stage):
```

Comment:
We'd like to avoid general samples if at all possible. Could you remove the arguments and use specific values?

Otherwise, provide example values as comments if not possible to hardcode values for these.